internal/cli/atlas_clusters_create.go

```diff
@@ -56,31 +62,40 @@ func (opts *atlasClustersCreateOpts) init() error {
 }
 
 func (opts *atlasClustersCreateOpts) Run() error {
-	cluster := opts.newCluster()
+	cluster, err := opts.newCluster()
+	if err != nil {
+		return err
+	}
 	result, err := opts.store.CreateCluster(cluster)
-
 	if err != nil {
 		return err
 	}
 
 	return json.PrettyPrint(result)
 }
 
-func (opts *atlasClustersCreateOpts) newCluster() *atlas.Cluster {
-	replicationSpec := opts.newReplicationSpec()
-	providerSettings := opts.newProviderSettings()
-
-	cluster := &atlas.Cluster{
-		BackupEnabled:       &opts.backup,
-		ClusterType:         replicaSet,
-		DiskSizeGB:          &opts.diskSizeGB,
-		GroupID:             opts.ProjectID(),
-		MongoDBMajorVersion: opts.mdbVersion,
-		Name:                opts.name,
-		ProviderSettings:    providerSettings,
-		ReplicationSpecs:    []atlas.ReplicationSpec{replicationSpec},
+func (opts *atlasClustersCreateOpts) newCluster() (*atlas.Cluster, error) {
+	cluster := new(atlas.Cluster)
+	if opts.filename != "" {
+		if err := file.Load(opts.fs, opts.filename, cluster); err != nil {
+			return nil, err
+		}
+	} else {
+		cluster.ClusterType = replicaSet
+		opts.applyOpts(cluster)
 	}
-	return cluster
+	cluster.GroupID = opts.ProjectID()
+	return cluster, nil
+}
+
+func (opts *atlasClustersCreateOpts) applyOpts(out *atlas.Cluster) {
+	replicationSpec := opts.newReplicationSpec()
```

Comment:
[q] Have options like `ClusterType` now moved into this option? I'm not seeing a 1:1 mapping with settings from the previous version of this file.

Reply:
ClusterType is now on line 84, I kind of split this in a way that if tomorrow we want to mix file and flags we just need to take this out of the `else`
This is me thinking if someone may have a file but no mdb version defined and then just do something like `--file file --mdb 4.2` right now is one or the other in sake of keeping it simple but I do see the use case for someone going the other way so this is a small future proof of it

Reply:
👍